File: Recursion/factorial.cpp
#include<iostream>
using namespace std;
int factorial(int n)
{
    if(n == 0)
    {
        return 1;
    }
    int ans = factorial(n - 1);
    return ans * n;
}

int main()
{
    int n;
    cin >> n;
    cout << factorial(n);
    return 0;
}

File: Recursion/factorial.java
import java.util.Scanner;
public class factorial
{
    public static Scanner scn = new Scanner(System.in);
    public static int fact(int n)
    {
        if(n == 0)
        {
            return 1;
        }
        int ans = fact(n - 1);
        return ans * n;
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        System.out.print(fact(n));
    }
}